src/test/java/com/worksap/fig/CharSeqTest.java

```diff
@@ -1,18 +1,190 @@
 package com.worksap.fig;
 
 import com.worksap.fig.lang.CharSeq;
+import com.worksap.fig.lang.Seq;
 import org.junit.Test;
 
-import static org.junit.Assert.assertEquals;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
 /**
- * Created by liuyang on 7/27/15.
+ * Created by lijunxiao on 7/27/15.
  */
 public class CharSeqTest {
+    @Test
+    public void testSubSeq() {
+        CharSeq seq = CharSeq.of("Father Charles gets down and ends battle.");
+        assertEquals(CharSeq.of("Father"), seq.subSeq(0, 6));
+        assertEquals(CharSeq.of("Charles "), seq.subSeq(7, 15));
+        assertEquals(CharSeq.of("Charles gets down and ends battle."), seq.subSeq(7, 41));
+        assertEquals(CharSeq.of("."), seq.subSeq(40));
+        assertEquals(CharSeq.of(""), seq.subSeq(41));
+        assertEquals(CharSeq.of("Father Charles gets down and ends battle."), seq.subSeq(0));
+    }
+
+    @Test
+    public void testLength() {
+        CharSeq seq = CharSeq.of("The quick brown fox jumps over a lazy dog");
+        assertEquals(41, seq.length());
+
+        CharSeq emptySeq = CharSeq.of("");
+        assertEquals(0, emptySeq.length());
+    }
+
+    @Test
+    public void testCapitalize() {
+        CharSeq seq = CharSeq.of("the quick brown fox jumps over a lazy dog");
+        assertEquals(CharSeq.of("The quick brown fox jumps over a lazy dog"), seq.capitalize());
+
+        assertEquals(CharSeq.of(""), CharSeq.of("").capitalize());
```

Comment:
The test can not show "upper case letter which is not the first letter should become lower case"